### ISSUE TITLE
`kubectl describe <TPR>` doesn't work

### ISSUE THREAD

**Issue by snarlysodboxer** [coreos/prometheus-operator#91]
Using Prometheus Operator v0.2.2:
```shell
david@machine:~/prometheus$ kubectl get servicemonitors,alertmanagers,prometheuses --all-namespaces
NAMESPACE   NAME                                    KIND
default     servicemonitors/prometheus-front-tier   ServiceMonitor.v1alpha1.monitoring.coreos.com

NAMESPACE   NAME                                    KIND
default     alertmanagers/alertmanager-front-tier   Alertmanager.v1alpha1.monitoring.coreos.com

NAMESPACE   NAME                                 KIND
default     prometheuses/prometheus-front-tier   Prometheus.v1alpha1.monitoring.coreos.com
david@machine:~/prometheus$
david@machine:~/prometheus$ kubectl describe prometheuses/prometheus-front-tier
the provided version "monitoring.coreos.com/v1alpha1" has no relevant versions: group monitoring.coreos.com has not been registered
no matches for monitoring.coreos.com/, Kind=Prometheus
david@machine:~/prometheus$
david@machine:~/prometheus$ kubectl describe alertmanagers/alertmanager-front-tier
the provided version "monitoring.coreos.com/v1alpha1" has no relevant versions: group monitoring.coreos.com has not been registered
no matches for monitoring.coreos.com/, Kind=Alertmanager
david@machine:~/prometheus$
david@machine:~/prometheus$ kubectl describe servicemonitors/prometheus-front-tier
the provided version "monitoring.coreos.com/v1alpha1" has no relevant versions: group monitoring.coreos.com has not been registered
no matches for monitoring.coreos.com/, Kind=ServiceMonitor
david@machine:~/prometheus$
david@machine:~/prometheus$ kubectl version
Client Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.1", GitCommit:"82450d03cb057bab0950214ef122b67c83fb11df", GitTreeState:"clean", BuildDate:"2016-12-14T00:57:05Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.1", GitCommit:"82450d03cb057bab0950214ef122b67c83fb11df", GitTreeState:"clean", BuildDate:"2016-12-14T00:52:01Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"linux/amd64"}
david@machine:~/prometheus$
```

**Comment by brancz** [comment]
I believe this is not supported for TPRs right now. 

> group monitoring.coreos.com has not been registered

It looks like this is an error that this group is not registered on the kubectl side, which unfortunately has no possibility of extension at this point. The solutions are still being discussed upstream.

**Comment by snarlysodboxer** [comment]
@brancz [this](http://kubernetes.io/docs/user-guide/kubectl/kubectl_describe/) seems to indicate is should work (under "Valid resource types include:",) but maybe it's wrong?

Also, [this](https://github.com/kubernetes/kubernetes/blob/master/docs/design/extending-api.md#expectations-about-third-party-objects) seems to indicate the problem may be missing [metadata](https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#metadata).

**Comment by AlexeyKasatkin** [comment]
Facing the same issue:

```
root@akasatkin-k8s-01:~# kubectl get ipclaim
NAME          KIND
10-0-0-7-24   IpClaim.v1.ipcontroller.ext
10-0-0-8-24   IpClaim.v1.ipcontroller.ext
10-0-0-9-24   IpClaim.v1.ipcontroller.ext
root@akasatkin-k8s-01:~# kubectl describe ipclaim
the provided version "ipcontroller.ext/v1" has no relevant versions: group ipcontroller.ext has not been registered
no matches for ipcontroller.ext/, Kind=IpClaim
root@akasatkin-k8s-01:~# kubectl edit ipclaim
the provided version "ipcontroller.ext/v1" has no relevant versions: group ipcontroller.ext has not been registered
no matches for ipcontroller.ext/, Kind=IpClaim
root@akasatkin-k8s-01:~# kubectl version
Client Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.1+coreos.0", GitCommit:"cc65f5321f9230bf9a3fa171155c1213d6e3480e", GitTreeState:"clean", BuildDate:"2016-12-14T04:08:28Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.1+coreos.0", GitCommit:"cc65f5321f9230bf9a3fa171155c1213d6e3480e", GitTreeState:"clean", BuildDate:"2016-12-14T04:08:28Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by brancz** [comment]
@AlexeyKasatkin as that is about the [k8s-externalipcontroller](https://github.com/Mirantis/k8s-externalipcontroller) problems with that would better be discussed in that repository.

I did some research through the Kubernetes repository but have not been very successful in finding an answer these are two of the errors I am getting through kubectl:

https://github.com/kubernetes/kubernetes/blob/6d9e2afedab933641833e9614bde362416072183/pkg/kubectl/cmd/util/factory_object_mapping.go#L180
https://github.com/kubernetes/kubernetes/blob/55bee3ad21f025b1416a4e1f10de753f484b66d3/staging/src/k8s.io/apimachinery/pkg/api/meta/restmapper.go#L533

I'm unsure where the error is as it seems there should at least be a general fallback that just displays the name, namespace and labels of an object when it is not a known Kubernetes object kind.

**Comment by AlexeyKasatkin** [comment]
@brancz Thank you. I've found a number of issued and PRs (mentioned in https://github.com/coreos/prometheus-operator/issues/118). Seems, it was fixed in k8s master. I'll check that.

**Comment by brancz** [comment]
This is indeed fixed in Kubernetes 1.6 :tada:
